.github/workflows/build_and_test.yml

```diff
@@ -32,6 +32,7 @@ jobs:
           apk update && apk add cppcheck python3-dev
           python3 -m pip install cmake-format clang-format==18.1.6
 
+      # TODO: investigate how to run pre-commit with `venv`
```

Comment:
Not related with this PR directly but, it is possible to use [pre-commit.ci](https://https://pre-commit.ci/) but we should remove cppcheck and clang-format from pre-commit since they require pre-installed binaries.